LiveSched/src/test/java/dev/coms4156/project/livesched/ScheduleUnitTests.java

```
@@ -23,145 +25,134 @@
+
+    mockResource1 = mock(Resource.class);
+    mockResource2 = mock(Resource.class);
+    mockResourceType = mock(ResourceType.class);
+
+    taskLocation = mock(Location.class);
+    resourceLocation = mock(Location.class);
+    when(taskLocation.getDistance(any())).thenReturn(50.0);
+    when(resourceLocation.getDistance(any())).thenReturn(50.0);
   }
 
   /**
    * Test for the constructor in Schedule class.
    */
   @Test
   void constructorTest() {
-    assertDoesNotThrow(() -> new Schedule(scheduleId, resourceTypes, maxDistance),
+    assertDoesNotThrow(() -> new Schedule(testScheduleId, mockTasks, maxDistance),
         "Schedule constructor should not throw an exception with valid parameters.");
 
-    Exception exception = assertThrows(IllegalArgumentException.class, 
-        () -> new Schedule(null, resourceTypes, maxDistance),
+    Exception exception = assertThrows(IllegalArgumentException.class,
+        () -> new Schedule(null, mockTasks, maxDistance),
         "Schedule constructor should throw an exception if scheduleId is null.");
     assertEquals("Schedule ID cannot be null or empty.", exception.getMessage());
 
-    exception = assertThrows(IllegalArgumentException.class, 
-        () -> new Schedule("  ", resourceTypes, maxDistance),
+    exception = assertThrows(IllegalArgumentException.class,
+        () -> new Schedule("  ", mockTasks, maxDistance),
         "Schedule constructor should throw an exception if scheduleId is empty.");
     assertEquals("Schedule ID cannot be null or empty.", exception.getMessage());
 
-    exception = assertThrows(IllegalArgumentException.class, 
-        () -> new Schedule(scheduleId, null, maxDistance),
-        "Schedule constructor should throw an exception if resourceTypes is null.");
-    assertEquals("Resource types list cannot be null.", exception.getMessage());
+    exception = assertThrows(IllegalArgumentException.class,
+        () -> new Schedule(testScheduleId, null, maxDistance),
```

Should we also consider a case where task list is empty? This also makes me think whether we should allow real-time processing of incoming task data. I think this could be done by using a separate thread that receives incoming tasks from the network and adds to the queue -- what do you think?

---

In that case, we won't need to throw an error when task list is empty. `createSchedule` also won't need to iterate through the task list but instead process task one by one whenever there is an element in the queue. We'd have to keep a separate thread for `createSchedule` as well.

---

Real time processing of incoming tasks sounds great! we can add it to the Jira tasks to address if/when we have time. I think createSchedule may still need to iterate if we want to create a schedule of multiple tasks mapped to resources, addressing only one at a time seems to beat the purpose of promoting efficiency through our service. But if I'm missing something there happy to understand better over discord!

---

Also, will add a test for empty task list in next PR